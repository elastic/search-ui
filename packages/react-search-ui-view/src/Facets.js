import PropTypes from "prop-types";
import React from "react";

function Facets({ children }) {
  return <div className="sui-facet-container">{children}</div>;
}

Facets.propTypes = {
  children: PropTypes.node.isRequired
};

export default Facets;
