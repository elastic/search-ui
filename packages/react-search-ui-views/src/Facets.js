import PropTypes from "prop-types";
import React from "react";

import { appendClassName } from "./view-helpers";

function Facets({ children, className, ...rest }) {
  return (
    <div
      className={appendClassName("sui-facet-container", className)}
      {...rest}
    >
      {children}
    </div>
  );
}

Facets.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Facets;
