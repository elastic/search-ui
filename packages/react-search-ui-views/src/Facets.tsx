import React from "react";

import { appendClassName } from "./view-helpers";

type FacetsProps = {
  children: React.ReactNode;
  className?: string;
} & React.HTMLAttributes<HTMLDivElement>;

function Facets({ children, className, ...rest }: FacetsProps) {
  return (
    <div
      className={appendClassName("sui-facet-container", className)}
      {...rest}
    >
      {children}
    </div>
  );
}

export default Facets;
