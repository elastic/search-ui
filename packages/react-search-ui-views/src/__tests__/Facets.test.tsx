import React from "react";
import Facets from "../Facets";
import { shallow } from "enzyme";

it("renders correctly", () => {
  const wrapper = shallow(
    <Facets>
      <div>Children</div>
    </Facets>
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const wrapper = shallow(
    <Facets className={customClassName}>
      <div>Children</div>
    </Facets>
  );
  const { className } = wrapper.props();
  expect(className).toEqual("sui-facet-container test-class");
});
