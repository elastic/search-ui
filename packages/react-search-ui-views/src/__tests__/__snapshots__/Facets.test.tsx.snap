// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sui-facet-container"
>
  <div>
    Children
  </div>
</div>
`;
